### PR TITLE
feat(UI): application data is now loaded in UI through the new GetAppDetails endpoint 

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**
Ref: SRX-FIC65Q

### CODE REVIEW

File: services/frontend-service/src/ui/components/ServiceLane/ServiceLane.scss [hunk 1]
```diff
@@ -62,6 +62,7 @@ Copyright freiheit.com*/
         display: flex;
         margin: $service-lane-releases-margin;
         flex-flow: wrap;
+        min-height: 85px;
```
Comment (miguel-crespo-fdc) [inline services/frontend-service/src/ui/components/ServiceLane/ServiceLane.scss:65]: Make sure that the loading service lane has the same height as the ready one

File: services/cd-service/pkg/service/overview.go [hunk 2]
```diff
@@ -217,14 +211,19 @@ func (o *OverviewServiceServer) GetAppDetails(
 			}
 			response.Deployments[envName] = deployment
 		}
+		result.UndeploySummary = deriveUndeploySummary(appName, response.Deployments)
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/service/overview.go:214]: Fixed bug on calculation of theses values

File: services/frontend-service/src/ui/Pages/ReleaseHistory/ReleaseHistoryPage.test.tsx [hunk 3]
```diff
@@ -81,7 +82,16 @@ describe('LocksPage', () => {
             if (testcase.enableDex) {
                 enableDexAuth(testcase.enableDexValidToken);
             }
+
+            updateAppDetails.set({
+                '': {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/Pages/ReleaseHistory/ReleaseHistoryPage.test.tsx:87]: why do we need an entry for ''?

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 4]
```diff
@@ -372,9 +372,14 @@ export const undeployTooltipExplanation =
 export const ReleaseDialog: React.FC<ReleaseDialogProps> = (props) => {
     const { app, className, version } = props;
     // the ReleaseDialog is only opened when there is a release, so we can assume that it exists here:
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:374]: this comment seems outdated, please update or remove

File: services/frontend-service/src/ui/components/ServiceLane/ServiceLane.scss (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/ServiceLane/ServiceLane.scss:65]: this should be a comment in the code
```suggestion
        min-height: 85px; // Ensure that the loading service lane has the same height as the ready one
```

File: services/frontend-service/src/ui/components/ServiceLane/ServiceLane.test.tsx [hunk 5]
```diff
@@ -280,15 +391,22 @@ describe('Service Lane Diff', () => {
                     },
                 ],
             });
-            const sampleApp: Application = {
-                undeploySummary: UndeploySummary.NORMAL,
+            // const sampleApp: Application = {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ServiceLane/ServiceLane.test.tsx:394]: please remove dead code

File: services/frontend-service/src/ui/components/ServiceLane/ServiceLane.tsx [hunk 6]
```diff
@@ -104,15 +104,61 @@ const deriveUndeployMessage = (undeploySummary: UndeploySummary): string | undef
...
+    hideMinors: boolean;
+    appDetails: GetAppDetailsResponse;
+}> = (props) => {
     const { application, hideMinors } = props;
-    const deployedReleases = useDeployedReleases(application.name);
-    const allReleases = useVersionsForApp(application.name);
     const { navCallback } = useNavigateWithSearchParams('releasehistory/' + application.name);
-    const prepareUndeployOrUndeployText = deriveUndeployMessage(application.undeploySummary);
+
+    const allReleases = [...new Set(props.appDetails?.application?.releases.map((d) => d.version))].sort(
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ServiceLane/ServiceLane.tsx:150]: could we instead change the server response to be already sorted? (nitpick)

File: services/frontend-service/src/ui/utils/store.tsx [hunk 7]
```diff
@@ -501,29 +553,72 @@ export const useTeamLocks = (): DisplayLock[] =>
 export const getPriorityClassName = (envOrGroup: Environment | EnvironmentGroup): string =>
     'environment-priority-' + String(Priority[envOrGroup?.priority ?? Priority.UNRECOGNIZED]).toLowerCase();
 
+export const useNewTeamLocks = (app: string, team: string): DisplayLock[] => {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/utils/store.tsx:556]: why "new"? Can we remove the old one and rename this to "useTeamLocks"?

File: services/frontend-service/src/ui/utils/store.tsx [hunk 8]
```diff
@@ -1017,19 +1111,32 @@ export const useCurrentlyExistsAtGroup = (application: string): EnvironmentGroup
...
+        return [];
+    } else {
+        return appDetails.application?.releases;
+    }
+};
 
-// Get all release versions for an app
-export const useVersionsForApp = (app: string): number[] => useReleasesForApp(app).map((rel) => rel.version);
+// // Get all release versions for an app
+// export const useVersionsForApp = (app: string): number[] => useReleasesForApp(app).map((rel) => rel.version);
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/utils/store.tsx:1124]: remove dead code

File: services/frontend-service/src/ui/utils/store.tsx [hunk 9]
```diff
@@ -1038,13 +1145,16 @@ export const useReleaseDifference = (toDeployVersion: number, application: strin
...
 };
 // Get all minor releases for an app
-export const useMinorsForApp = (app: string): number[] =>
-    useReleasesForApp(app)
-        .filter((rel) => rel.isMinor)
+export const useMinorsForApp = (app: string): number[] | undefined =>
+    useAppDetailsForApp(app)
+        .application?.releases.filter((rel) => rel.isMinor)
         .map((rel) => rel.version);
+// .application?.releases.filter((rel) => rel.isMinor)
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/utils/store.tsx:1156]: remove dead code

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: pkg/api/v1/api.proto [hunk 10]
```diff
@@ -434,12 +434,19 @@ message GetOverviewRequest {
   string git_revision = 1;
 }
 
+//Lightweight version of application. Only contains name, team and warnings
```
Comment (sven-urbanski-freiheit-com) [inline pkg/api/v1/api.proto:437]: I do not see warnings here
```suggestion
//Lightweight version of application. Only contains name and team.
```

File: services/frontend-service/pkg/handler/commit_deployments_test.go [hunk 11]
```diff
@@ -83,7 +83,7 @@ func TestHandleCommitDeployments(t *testing.T) {
 			inputTail:          "123456/",
 			failGrpcCall:       false,
 			expectedStatusCode: http.StatusOK,
-			expectedResponse:   "{\"deploymentStatus\":{\"app1\":{\"deploymentStatus\":{\"dev\":\"DEPLOYED\",\"prod\":\"UNKNOWN\",\"stage\":\"PENDING\"}}}}\n",
+			expectedResponse:   "{\"deploymentStatus\":{\"app1\":{\"deploymentStatus\":{\"dev\":\"DEPLOYED\", \"prod\":\"UNKNOWN\", \"stage\":\"PENDING\"}}}}\n",
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/handler/commit_deployments_test.go:86]: is this really correct? Why are there spaces now?

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/frontend-service/pkg/handler/commit_deployments_test.go (hunk 11 above)
Comment (miguel-crespo-fdc) [reply services/frontend-service/pkg/handler/commit_deployments_test.go]: This is a flakey test . This version runs on the pipeline but does not run locally.

**Pull request merged**